### PR TITLE
feat(NN): add MsFFN, PFNN, STMsFFN in paddle

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
from PR https://github.com/lululxvi/deepxde/pull/1070

1. add MsFFN, PFNN, STMsFFN in paddle

### CODE REVIEW

File: deepxde/nn/paddle/fnn.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
 import paddle
 
+from .. import activations, initializers
```
Comment (lululxvi) [inline deepxde/nn/paddle/fnn.py:3]: Use the original code, so the import style is consistent with other codes in DeepXDE.

File: deepxde/nn/paddle/fnn.py [hunk 2]
```diff
@@ -30,3 +30,111 @@ def forward(self, inputs):
...
+
+    Args:
+        layer_sizes: A nested list that defines the architecture of the neural network
+            (how the layers are connected). If `layer_sizes[i]` is an int, it represents
+            one layer shared by all the outputs; if `layer_sizes[i]` is a list, it
+            represents `len(layer_sizes[i])` sub-layers, each of which is exclusively
+            used by one output. Note that `len(layer_sizes[i])` should equal the number
+            of outputs. Every number specifies the number of neurons in that layer.
+        activation: If `activation` is a ``string``, then the same activation is used in
+            both trunk and branch nets. If `activation` is a ``dict``, then the trunk
```
Comment (lululxvi) [inline deepxde/nn/paddle/fnn.py:47]: No trunk and branch nets.

File: deepxde/nn/paddle/fnn.py [hunk 3]
```diff
@@ -30,3 +30,111 @@ def forward(self, inputs):
...
+            net uses the activation `activation["trunk"]`, and the branch net uses
+            `activation["branch"]`.
+        kernel_initializer: initializer method for NN kernels.
```
Comment (lululxvi) [inline deepxde/nn/paddle/fnn.py:50]: kernel_initializer: Initializer for the kernel weights matrix.

File: deepxde/nn/paddle/msffn.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        self,
+        layer_sizes,
+        activation,
+        kernel_initializer,
+        sigmas,
+        dropout_rate=0.0,
+        use_bias=True
+    ):
+        super().__init__()
+        self._input_transform = None
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:31]: This is already in the base class `NN`

File: deepxde/nn/paddle/msffn.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    def _fully_connected_forward(self, y):
+        for i in range(len(self.linears)):
+            y = self.linears[i](y)
+            y = self.activation(y)
+            if self.dropout_rate > 0:
+                y = paddle.nn.functional.dropout(
+                    y, p=self.dropout_rate, training=self.training)
+        return y
+
+    def apply_feature_transform(self, transform):
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:107]: This is already in the base class `NN`

File: deepxde/nn/paddle/msffn.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        """Compute the features by appling a transform to the network inputs, i.e.,
+        features = transform(inputs). Then, outputs = network(features).
+        """
+        # TODO: support input transform
+        self._input_transform = transform
+
+    def apply_output_transform(self, transform):
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:114]: This is already in the base class `NN`

File: deepxde/nn/paddle/msffn.py [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+        self.activation = activations.get(activation)
+        self.dropout_rate = dropout_rate
+        self.sigmas = sigmas  # list or tuple
+        self.use_bias = use_bias
+        self.fourier_feature_weights = None
+        initializer = initializers.get(kernel_initializer)
+        initializer_zero = initializers.get("zeros")
+
+        self.b = []
+        for i, sigma in enumerate(self.sigmas):
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:43]: `i` is never used.

File: deepxde/nn/paddle/msffn.py [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    """
+
+    def __init__(
+        self,
+        layer_sizes,
+        activation,
+        kernel_initializer,
+        sigmas,
+        dropout_rate=0.0,
+        use_bias=True
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:28]: `use_bias` is never used. Delete.

File: deepxde/nn/paddle/msffn.py [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+    def __init__(
+        self,
+        layer_sizes,
+        activation,
+        kernel_initializer,
+        sigmas_x,
+        sigmas_t,
+        dropout_rate=0,
+        use_bias=True
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:140]: Delete.

File: deepxde/nn/paddle/fnn.py (hunk 2 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/fnn.py]: Corrected

File: deepxde/nn/paddle/fnn.py (hunk 3 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/fnn.py]: Done

File: deepxde/nn/paddle/msffn.py (hunk 4 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py]: Done

File: deepxde/nn/paddle/msffn.py (hunk 5 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py]: Deleted

File: deepxde/nn/paddle/msffn.py (hunk 6 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py]: Deleted

File: deepxde/nn/paddle/msffn.py (hunk 7 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py]: Deleted

File: deepxde/nn/paddle/msffn.py (hunk 8 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py:28]: Previously omitted `use_bias` has been added in `self._dense` as an arg

Comment (lululxvi) [reply deepxde/nn/paddle/msffn.py:28]: Let us remove `use_bias`. It is not really useful.

File: deepxde/nn/paddle/msffn.py [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    ):
+        layer_sizes[0] = layer_sizes[0] // 2
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:123]: `//2` may not be correct. `x` could be high dim, such as x, y ,z .

File: deepxde/nn/paddle/msffn.py [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+        self.sigmas_x = sigmas_x
+        self.sigmas_t = sigmas_t
+
+    def forward(self, inputs):
+        x = inputs
+        if self._input_transform is not None:
+            # The last column should be function of t.
+            x = self._input_transform(x)
+
+        x0, x1 = paddle.split(x, num_or_sections=2, axis=1)
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:141]: Also, here `num_or_sections=2`.

File: deepxde/nn/paddle/msffn.py (hunk 10 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py]: Done

File: deepxde/nn/paddle/msffn.py (hunk 11 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py]: Done

File: deepxde/nn/paddle/msffn.py [hunk 12]
```diff
@@ -0,0 +1,184 @@
+from .nn import NN
+from .. import activations
+from .. import initializers
+from ...backend import paddle
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:4]: Directly
```
import paddle
```

File: deepxde/nn/paddle/msffn.py [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+        113938, 2021 <https://doi.org/10.1016/j.cma.2021.113938>`_.
+    """
+
+    def __init__(
+        self,
+        layer_sizes,
+        activation,
+        kernel_initializer,
+        sigmas,
+        dropout_rate=0.0,
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:27]: `dropout_rate=0`, to be consistent with STMsFFN

File: deepxde/nn/paddle/msffn.py (hunk 12 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py]: Done

File: deepxde/nn/paddle/msffn.py (hunk 13 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py]: Done

File: deepxde/nn/paddle/msffn.py [hunk 14]
```diff
@@ -0,0 +1,184 @@
+from .nn import NN
+from .. import activations
+from .. import initializers
+import paddle
```
Comment (lululxvi) [inline deepxde/nn/paddle/msffn.py:4]:
```
import paddle

from .nn import NN
from .. import activations
from .. import initializers
```

Comment (HydrogenSulfate) [reply deepxde/nn/paddle/msffn.py]: Done
